File: UCA-4/Arrays/C++/rotateArray.cpp
/*
* Rotate An Array By Given Times
*/
#include <iostream>
using namespace std;
int size = 0;

void rotate(int *arr,int pos=0){
	while(pos--){
		int temp = arr[size-1];
		for (int i = size-1; i > 0; --i)
		{
			arr[i] = arr[i-1];
		}
		arr[0] = temp;
	}
}

void traverse(int *arr){
	for (int i = 0; i < size; ++i)
	{
		cout<<arr[i]<<" ";
	}
	cout<<endl;
}

int main(int argc, char const *argv[])
{
	size = atoi(argv[1]);
	int arr[size];
	for (int i = 0; i < size; ++i)
	{
		arr[i] = atoi(argv[i+2]);	
	}
	rotate(arr,1);
	traverse(arr);
	return 0;
}

File: UCA-4/Arrays/Java/rotateArray.java
/*
* Rotate An Array By Given Times
*/
import java.util.*;

class test{

	static void rotateArray(int[] arr,int pos){
		while((pos--)!=0){
			int temp = arr[arr.length-1];
			for (int i = arr.length-1; i > 0; --i)
			{
				arr[i] = arr[i-1];
			}
			arr[0] = temp;
		}
	}

	public static void main(String[] args) {
		int size = Integer.parseInt(args[0]);
		int[] arr = new int[size];
		for (int i=0;i<size;i++) {
			arr[i] = Integer.parseInt(args[i+1]);
		}
		rotateArray(arr,2);
		for (int x :arr) {
			System.out.println(x);
		}
	}
}